src/main/java/com/devcycle/sdk/server/local/api/DVCLocalClient.java

```diff
@@ -113,40 +110,43 @@ public <T> Variable<T> variable(User user, String key, T defaultValue) {
             .isDefaulted(true)
             .build();
 
-    if (!isInitialized) {
+    if (!configManager.isConfigInitialized() || !isInitialized) {
+      System.out.println("Variable called before DVCClient has initialized, returning default value");
+      try {
+        eventQueueManager.queueAggregateEvent(Event.builder().type("aggVariableDefaulted").target(key).build(), null);
+      } catch (Exception e) {
+        System.out.printf("Unable to parse aggVariableDefaulted event for Variable %s due to error: %s", key, e.toString());
+      }
       return defaultVariable;
     }
-
+    String variableJSON = null;
     try {
-      BucketedUserConfig bucketedUserConfig = localBucketing.generateBucketedConfig(sdkKey, user);
-      if (bucketedUserConfig.variables.containsKey(key)) {
-        BaseVariable baseVariable = bucketedUserConfig.variables.get(key);
+      variableJSON = localBucketing.getVariable(sdkKey, user, key, variableType, true);
+      if (variableJSON == null || variableJSON.isEmpty()) {
+        return defaultVariable;
+      } else {
+        ObjectMapper mapper = new ObjectMapper();
+        Variable baseVariable = mapper.readValue(variableJSON, Variable.class);
+
         Variable<T> variable = (Variable<T>) Variable.builder()
```

Comment:
Seems non-optimal that we are initializing 2 `Variable` objects here. Probably not something we need to optimize right now, but we should figure out how to only generate it once.